scripts/get_orphaned_functions.py

```diff
@@ -0,0 +1,249 @@
+    "\033[92m",  # BRIGHT_GREEN-10
+    "\033[93m",  # BRIGHT_YELLOW-11
+    "\033[94m",  # BRIGHT_BLUE-12
+    "\033[95m",  # BRIGHT_MAGENTA-13
+    "\033[96m",  # BRIGHT_CYAN-14
+    "\033[97m",  # BRIGHT_WHITE-15
+]
+
+# Reset
+ENDC = "\033[0m"
+
+# List of modules to check for orphaned functions
+# Modules Not included : data.hydrology, DART due to differing file structures. Script should be enhanced further to remove this limitation.
+module_list = [
+    "assim.batch",
+    "assim.sequential",
+    "data.remote",
+    "photosynthesis",
+    "uncertainty",
+    "allometry",
+    "benchmark",
+    "data.land",
+    "emulator",
+    "priors",
+    "data.atmosphere",
+    "meta.analysis",
+    "rtm",
+]
+
+ignored_modules = ["data.hydrology", "DART"]
+
+# Initialize two lists to store the orphaned functions
+# List_1: Purely orphaned functions
+# List_2: Functions that are utilized externally
+list_1 = []
+list_2 = []
+
+main_dir = os.getcwd()
+
+
+# Define a function to list all functions in a module
+def list_functions_in_module(module_path):
+    """
+    List all functions defined in the R files of a module.
+
+    Args:
+    module_path (str): The path to the module's R directory.
+
+    Returns:
+    list: A list of function names defined in the module.
+    """
+    function_pattern = re.compile(r"\b([a-zA-Z_]\w*(?:\.\w+)*)\s*<-\s*function")
+    function_names = []
+
+    # Get all R files in the module_path
+    r_files = [f for f in os.listdir(module_path) if f.endswith(".R")]
+
+    # List all R files in the module_path
+    # print(color_codes[5] + "R files in the module_path of \t",color_codes[9],module_path," are \n ",color_codes[3],r_files,"\n",ENDC,)
+
+    # Findings all function names in the R files
+    for r_file in r_files:
+        with open(os.path.join(module_path, r_file), "r") as file:
+            content = file.read()
+            matches = function_pattern.findall(content)
+            function_names.extend(matches)
+
+    function_names = sorted(set(function_names))
+
+    # Print the function names we extracted and their count
+    # print(color_codes[13] + "Function names:", function_names, "\n", ENDC)
+    # print(color_codes[2] + "Length of function List", len(function_names), "\n", ENDC)
+
+    return function_names
+
+
+# Define a function to check if a function is utilized within its module
+def is_function_utilized_within_module(function_name, module_r_path):
+    """
+    Check if the function is utilized within its module.
+
+    Args:
+    function_name (str): The name of the function to check.
+    module_r_path (str): The path to the module's R directory.
+
+    Returns:
+    bool: True if the function is utilized within the module, False otherwise.
+    """
+
+    # Regex pattern to detect usage of the function
+    function_pattern = re.compile(r"\b" + re.escape(function_name) + r"\b\s*\(")
+
+    # Traverse all files in the given module R directory
+    for r_file_name in os.listdir(module_r_path):
```

Comment:
Updated file as per requirements in a test branch [here](https://github.com/Sweetdevil144/pecan/blob/orphan-test-branch/scripts/get_orphaned_functions.py). Can you verify if this is correctly performed?

Reply:
@Sweetdevil144 Please go ahead and commit those changes to the existing PR instead of leaving them in another branch -- it takes a fair number of extra steps for us to go find the diffs between your branches, and once we've found it there's no easy UI for us to leave comments on the changes. If instead you commit it to the existing `develop-ophanedFunctions` branch that this PR is using, then they'll show up here and we can easily discuss them in context.

Reply:
@infotroph Pushed. Can you review the changes once?